packages/contracts/contracts/DependencyRegistryV0.sol

```diff
@@ -742,7 +821,20 @@ contract DependencyRegistryV0 is
             return "";
         }
 
-        return _readFromBytecode(dependency.scriptBytecodeAddresses[_index]);
+        address scriptAddress = dependency.scriptBytecodeAddresses[_index];
+        bytes32 storageVersion = BytecodeStorageReader
+            .getLibraryVersionForBytecode(scriptAddress);
+
+        if (storageVersion == BytecodeStorageReader.UNKNOWN_VERSION_STRING) {
+            return
+                string(
+                    BytecodeStorageReader.readBytesFromSSTORE2Bytecode(
+                        scriptAddress
+                    )
```

Comment:
for gzip specifically, we can just assume it is base64 encoded since we know it is EthFS, but if other libraries were uploaded to EthFS that made sense to use, we would need a way to specify their encoding. We can defer recording encoding for now as well, since this is an upgradeable contact, and we currently aren't planning on pointing to EthFS for any dependency data at the moment, IIUC.

Reply:
Hrm – this one is interesting to consider, will try to throw some time on for you + @yoshiwarab and I to talk about this a bit more synchronously.

That said, my inclination is that if we do want to go this route, I'd approach this in a follow up PR rather than updating this one with additional handling logic (though I will add the dev comment now).

Leaving this comment unresolved for now and will throw some time for us 3 onto the calendar.

Reply:
Per our conversation today synchronously – I think we treat this as an area of follow up work, but decouple that from this initial PR. 

I've filed this issue (https://github.com/ArtBlocks/artblocks-contracts/issues/911) as one for myself to follow up with on an additional PR for you two to review @yoshiwarab @ryley-o.